src/Frontend/src/views/throughputreport/setup/ConnectionSetupView.vue

```diff
@@ -20,7 +20,7 @@ onMounted(async () => {
 
 <template>
   <div class="row">
-    <p v-if="(settingsInfo?.broker_settings.length ?? 0 > 0) || (!isBrokerTransport && useIsMonitoringEnabled())">
+    <p v-if="(settingsInfo?.broker_settings.length ?? 0 > 0) || ((settingsInfo?.monitoring_settings.length ?? 0 > 0) && useIsMonitoringEnabled()) || (settingsInfo?.service_control_settings.length ?? 0 > 0)">
```

Comment:
This logic is a bit hard for me to parse (especially with the `.monitoring_settings` and `.service_control_settings` differences which looks very similar to the rest to the code).

Is it possible to put this behind a named function to make it more readable?

`v-if="shouldDisplayCollectionInstructions()` or something like that?

Reply:
I have refactor this, thanks for pointing it out